### DISCUSSION TITLE
New setup with T41U5XBB v2.09 - no motor movement

### DISCUSSION THREAD

**Discussion by isfl** [phil-barrett/grblHAL-teensy-4.x#92] [Q&A]
Hi there,
we just got a new board and trying to replace a CNC760 from EdingCNC. Currently bench testing and also tried connecting to stepper drives inside the old cabinet. I get no motion from motors, some very low rattle/hum but essentially no motion.

I took off the board from the cabinet and bench testing with a TMC2209 stepper stick, no motion. I am confused about the stepper drive wiring which seems really strange to me and there is not much wiring documentation out there for the Teensy bob.

So first question - our board was ordered with Ethernet, but print on the board still has "U" instead of "E" - is this normal ?
Second question and the core of the issue:
The Teensy bob from Phil has 4 pins for each axis, Gnd/Step/Dir/En.  The TMC2209 stepper stick has Dir/Step/En. Which should go to which ? There is a wiring diagram in one of the grblhal resources showing a random stepper drive that has 6 input pins and how to wire it. Well, the TMC2209 has just Stp/Dir input so how to wire it to the Teensy bob ? Furthermore, the main machine runs ESD808 drives and I used the same corresponding pins going to the old motion controller to the corresponding Step/Dir and En(5v?) to the Teensy board - no motion.

I feel there is a documentation void or lack of understanding on my part, the stepper drive convention is not used with the Teensy bob and it's getting me confused. By convention I mean, most stepper drives have En+/En-/Pul+/Pul-/Dir+/Dir- inputs.

For the TMC2209 stepstick I wired Teensy Dir/Step to TMC2209 Dir/Step, En to Gnd on the stepstick as instructed by it's doco, but no motion.

On the actual machine, a Blackbox brain board (8bit) can successfully control the ESD808 hybrid stepper drives by connecting Step>Step, Dir>Dir>, Pulse+>5Vreg. Wiring the Teensy bob the same way results in no motion.

On the Teensy bob, using ioSender, I can turn spindle on and see the red led. I can jog axis and check voltage on the axis output pins with a MM. I can see the voltage doing stuff so steppers should move. Maybe I am confused about the pin correspondence. 

Could we get some help wiring this board ?

**Comment by isfl** [comment]
Please disregard :) 
I inverted enable pins and now there is motion. Hurray

**Comment by phil-barrett** [comment]
Glad you got it worked out!  I have a stepper hook-up guide in the PicoCNC manual that I think is better and will move it over to the T41U5XBB.

The U vs E thing is from when I thought to have 2 different boards, one Ethernet and one only for USB.  It turned out easier to just have one board for both but the name stuck.
